src/rpc/rpcevo.cpp

```diff
@@ -811,21 +811,29 @@ UniValue bls_generate(const JSONRPCRequest& request)
     return ret;
 }
 
-UniValue _bls(const JSONRPCRequest& request)
+void _bls_help()
 {
-    if (request.params.empty()) {
-        throw std::runtime_error(
-                "bls \"command\" ...\n"
-                "Set of commands to execute BLS related actions.\n"
-                "To get help on individual commands, use \"help bls command\".\n"
-                "\nArguments:\n"
-                "1. \"command\"        (string, required) The command to execute\n"
-                "\nAvailable commands:\n"
-                "  generate          - Create a BLS secret/public key pair\n"
-        );
-    }
+    throw std::runtime_error(
+            "bls \"command\" ...\n"
+            "Set of commands to execute BLS related actions.\n"
+            "To get help on individual commands, use \"help bls command\".\n"
+            "\nArguments:\n"
+            "1. \"command\"        (string, required) The command to execute\n"
+            "\nAvailable commands:\n"
+            "  generate          - Create a BLS secret/public key pair\n"
+            );
+}
 
-    std::string command = request.params[0].get_str();
+UniValue _bls(const JSONRPCRequest& request)
+{
+    std::string command;
+    if (request.params.size() >= 1)
+        command = request.params[0].get_str();
+
+    if ((request.fHelp && command.empty())  ||
+        (
+         command != "generate" ))
```

Comment:
This kind of duplicates the `if/else` that follows later. Doesn't look that bad as of now, but will get bad when we add more and more commands. Suggestion:

```
    if (request.params.empty()) {
        _bls_help();
    }

    if (command == "generate") {
        bls_generate(request);
    } else if (...) {
        ....
    } else {
        _bls_help();
    }
```

Reply:
Agree, I just duplicated what `gobject` does. It seems we have a lack of inconsistency in how these things are done and I am not really sure which is more helpful - displaying help info or just clearly stating an invalid command was used.